content/automate/ManageIQ/PhysicalInfrastructure/PhysicalServer/Provisioning/StateMachines/Provision.class/__methods__/update_provision_status.rb

```diff
@@ -0,0 +1,71 @@
+#
+# Description: This method updates the service provisioning status
+# Required inputs: status
+#
+module ManageIQ
+  module Automate
+    module PhysicalInfrastructure
+      module PhysicalServer
+        module Provisioning
+          module StateMachines
+            module Provision
+              class UpdateProvisionStatus
+                def initialize(handle = $evm)
+                  @handle = handle
+                end
+
+                def task
+                  @task ||= begin
+                    task = @handle.root['miq_provision_task']
+                    unless task
+                      @handle.log(:error, 'miq_provision_task object not provided')
+                      exit(MIQ_STOP)
+                    end
+                    unless task.source
+                      @handle.log(:error, 'miq_provision_task.source object not provided (PhysicalServer expected)')
+                      exit(MIQ_STOP)
+                    end
+                    task
+                  end
```

Comment:
```
@task ||= @handle.root['miq_provision_task'].tap do |task|
  unless task
    @handle.log(:error, 'miq_provision_task object not provided')
    exit(MIQ_STOP)
  end
  unless task.source
    @handle.log(:error, 'miq_provision_task.source object not provided (PhysicalServer expected)')
    exit(MIQ_STOP)
  end
end
```

Reply:
You know, it felt like there is a better way, thanks 👍